File: news_server/modules/source/infrastructure/getters/toutiao.ts
/**
 * 今日头条新闻源获取器
 * 从 newsnow/server/sources/toutiao.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'

interface ToutiaoResponse {
  data: {
    ClusterIdStr: string
    Title: string
    HotValue: string
    Image: {
      url: string
    }
    LabelUri?: {
      url: string
    }
  }[]
}

export const toutiaoGetter: SourceGetter = async () => {
  const url = 'https://www.toutiao.com/hot-event/hot-board/?origin=toutiao_pc'
  
  const response = await fetch(url)
  if (!response.ok) {
    throw new Error(`Failed to fetch Toutiao: ${response.statusText}`)
  }

  const res = (await response.json()) as ToutiaoResponse
  
  return res.data.map((k) =>
    NewsItem.fromEntity({
      id: k.ClusterIdStr,
      title: k.Title,
      url: `https://www.toutiao.com/trending/${k.ClusterIdStr}/`,
      extra: {
        icon: k.LabelUri?.url
          ? {
              url: k.LabelUri.url,
              scale: 1,
            }
          : false,
      },
    })
  )
}

File: modules/source/infrastructure/getters/toutiao.go
package getters

import "context"

func (r *Registry) toutiao(ctx context.Context) ([]Item, error) {
	var res struct {
		Data []struct {
			ClusterIDStr string `json:"ClusterIdStr"`
			Title        string `json:"Title"`
		} `json:"data"`
	}
	if err := r.getJSON(ctx, "https://www.toutiao.com/hot-event/hot-board/?origin=toutiao_pc", nil, &res); err != nil {
		return nil, err
	}
	out := make([]Item, 0, len(res.Data))
	for _, k := range res.Data {
		out = append(out, Item{ID: k.ClusterIDStr, Title: k.Title, URL: "https://www.toutiao.com/trending/" + k.ClusterIDStr + "/"})
	}
	return out, nil
}
